public/app/features/trails/MetricScene.tsx

```diff
@@ -53,12 +54,13 @@ export class MetricScene extends SceneObjectBase<MetricSceneState> {
 
   public constructor(state: MakeOptional<MetricSceneState, 'body' | 'autoQuery' | 'layout'>) {
     const autoQuery = state.autoQuery ?? getAutoQueriesForMetric(state.metric);
+    const layout = localStorage.getItem(TRAIL_BREAKDOWN_VIEW_KEY) ?? 'grid';
     super({
       $variables: state.$variables ?? getVariableSet(state.metric),
       body: state.body ?? new MetricGraphScene({}),
       autoQuery,
       queryDef: state.queryDef ?? autoQuery.main,
-      layout: state.layout ?? 'grid',
+      layout: layout as LayoutType,
```

Comment:
I don't like the type casting but I couldn't find any other better way. I am open suggestions

Reply:
Update `ActionTabs/types.ts` to have the following:

```ts
const LAYOUT_TYPES = ['single', 'grid', 'rows'] as const;

export type LayoutType = typeof LAYOUT_TYPES[number];

export function isLayoutType(layoutType: string | null | undefined): layoutType is LayoutType {
    return !!layoutType && layoutType in LAYOUT_TYPES;
}
```


```suggestion
      layout: isLayoutType(layout) ? layout : 'grid',
```

And remove the `?? 'grid';` portion of `layout`'s definition above.